msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -0,0 +1,49 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Xamarin.iOS.Tasks {
+	[TestFixture]
+	public class LocalizationStringTest : TestBase {
+		[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
+		[TestCase ("de-DE", "Unbekanntes Imageformat.")]
+		[TestCase ("en-US", "Unknown image format.")]
+		[TestCase ("es-ES", "formato de imagen desconocido.")]
+		[TestCase ("fr-FR", "format d'image inconnu.")]
+		[TestCase ("it-IT", "Formato immagine sconosciuto.")]
+		[TestCase ("ja-JP", "の読み込みでエラーが発生しました: 画像の形式が不明です。")]
+		[TestCase ("ko-KR", "을(를) 로드하는 동안 오류 발생: 알 수 없는 이미지 형식입니다.")]
+		[TestCase ("pl-PL", "nieznany format obrazu.")]
+		[TestCase ("pt-BR", "formato de imagem desconhecido.")]
+		[TestCase ("ru-RU", "неизвестный формат изображения.")]
+		[TestCase ("tr-TR", "yüklenirken hata oluştu: Görüntü biçimi bilinmiyor.")]
+		[TestCase ("zh-CN", "时出错: 未知图像格式")]
+		[TestCase ("zh-TW", "時發生錯誤: 未知的映像格式。")]
+		public void AllSupportedTranslations (string culture, string errorMessage)
+		{
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+			CultureInfo newCulture;
+			try {
+				newCulture = new CultureInfo (culture);
+				Thread.CurrentThread.CurrentUICulture = newCulture;
+			}
+			catch (System.Globalization.CultureNotFoundException) {
```

Comment:
So, one tiny comment, if we are doing that, we could instead of a try/catch Assert.IsTrue do Assert.DoesNotThrow:
```csharp
Assert.DoesNotThrow (() => {
    newCulture = new CultureInfo (culture);
    Thread.CurrentThread.CurrentUICulture = newCulture;
}, $"{culture}: is not a valid culture.");
```

Reply:
With this, I wouldn't be able to set the original culture back if I hit the CultureNotFoundException though right?

Reply:
Using Assert.DoesNotThrow is never needed, it's just a very explicit way to say that you're verifying that a particular piece of code doesn't throw. The test works just as well without it.